build-system/babel-plugins/babel-plugin-transform-log-methods/index.js

```diff
@@ -0,0 +1,288 @@
+    fs.mkdirSync(messagesPathDir);
+  }
+  fs.writeFileSync(messagesPath, json);
+}
+
+/**
+ * Gets a message id from the table, or adds a new entry for a message if
+ * non-existent and returns its new id.
+ * @param {Object<string, string>} messages
+ * @param {string} message
+ * @param {function():number} getMessageId
+ * @return {string} Short message id.
+ */
+function getOrCreateShortMessageId(messages, message, getMessageId) {
+  if (messages[message]) {
+    return messages[message];
+  }
+  // Base-62 radix for best utilization of ascii space.
+  const shortMessageId = base62.encode(getMessageId());
+  messages[message] = shortMessageId;
+  return shortMessageId;
+}
+
+/**
+ * Builds a message template using printf syntax from a starting node.
+ * @param {@babel/types} t
+ * @param {!Node} node
+ * @param {!Array<!Node>} interpolationArgs
+ * @return {string} Template for the message.
+ */
+function buildMessage(t, node, interpolationArgs) {
+  if (t.isStringLiteral(node)) {
+    return node.value;
+  }
+
+  if (t.isBinaryExpression(node, {operator: '+'})) {
+    return (
+      buildMessage(t, node.left, interpolationArgs) +
+      buildMessage(t, node.right, interpolationArgs)
+    );
+  }
+
+  if (t.isTemplateLiteral(node)) {
+    let quasied = '';
+    let i = 0;
+    for (; i < node.quasis.length - 1; i++) {
+      quasied += node.quasis[i].value.cooked;
+      quasied += buildMessage(t, node.expressions[i], interpolationArgs);
+    }
+    quasied += node.quasis[i].value.cooked;
+    return quasied;
+  }
+
+  interpolationArgs.push(node);
+  return '%s';
+}
+
+/**
+ * @param {@babel/types} t
+ * @param {!Node} node
+ * @return {?Object}
+ */
+function getTransformableCallExpressionMeta(t, {callee}) {
+  // devAssert() or userAssert() call.
+  if (assertAliases.some(name => t.isIdentifier(callee, {name}))) {
+    return transformableMethods.assert;
+  }
+  // Looks like a method().call().
+  if (!t.isMemberExpression(callee) || !t.isCallExpression(callee.object)) {
+    return null;
+  }
+  // is either dev() or user() object.
+  const singletonCallee = callee.object.callee;
+  if (
+    !singletonFunctions.some(name => t.isIdentifier(singletonCallee, {name}))
+  ) {
+    return null;
+  }
+  // object property is transformable method.
+  const {property} = callee;
+  if (!t.isIdentifier(property) || !(property.name in transformableMethods)) {
+    return null;
+  }
+  return transformableMethods[property.name];
+}
+
+module.exports = function({types: t}) {
+  return {
+    /** Resolves plugin config. */
+    pre() {
+      const {
+        // Configurable to isolate test output.
+        messagesPath = defaultMessagesPath,
+
+        // Temporary option to not replace call arguments, but still output the
+        // table to keep build code and infra independent from rollout.
+        replaceCallArguments = true,
+      } = this.opts;
+
+      this.messagesPath = relativeToRoot(messagesPath);
```

Comment:
Don't store properties on onto `this`, it's actually the traversal context and not your plugin. Keep them as plugin scoped variables.